src/main/java/com/google/sps/servlets/OutputDocServlet.java

```diff
@@ -0,0 +1,20 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import com.google.gson.Gson;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/output-doc")
+public class OutputDocServlet extends HttpServlet {
+
+  // TODO: Load signed url download link from Cloud Storage
+  // Need to figure out way to pass SessionKeyId to this servlet
```

Comment:
@KevinWorkman How would we pass the SessionKeyId from the `FormHandlerServlet` to this servlet so that the output file link can be queried from the `Session` entity in Datastore? 

Sorry if the question is unclear!

Reply:
Should I be looking into servlet chaining or am I overcomplicating this?

Reply:
To check my understanding, does the flow work like this?

1. The user uploads an image, some processing happens, and `FormHandlerServlet` stores that in Datastore.
2. The user is redirected to output.html, which will display the output. (Looks like this part is still todo.)
3. This `OutputDocServlet` will return a doc, or a URL to a doc, which is added to the output.

Is all of that right? If so, then I think there's a more general question of how `output.html` knows which output to show?

You could use something like a URL parameter, something like `output.html?id=123` which the JavaScript then uses to request data from the server? Or you could use JSP to build the output on the server instead of using an HTML file?

I'm happy to discuss pros and cons over chat!

Reply:
Another method you could probably use would be servlet attributes and store the sessionID that way (https://www.studytonight.com/servlet/attribute.php). However, this might be a temporary fix for a MVP and not so scalable as it can only store one at a time

Reply:
I see those are some good ideas! Will definitely look into them. Thanks everyone!